components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ThrottlingApiServiceImpl.java

```diff
@@ -907,6 +997,323 @@ public Response throttlingPoliciesCustomRuleIdDelete(String ruleId,  MessageCont
+                    }
+                } catch (APIManagementException e) {
+                    String errorMessage = "Error while retrieving Custom policy: " + policyName;
+                    RestApiUtil.handleInternalServerError(errorMessage, e, log);
+                }
+            }
+            RestApiUtil.handleResourceNotFoundError("No throttle policy found by the name " + policyName, log);
+        } catch (APIManagementException | ParseException e) {
+            String errorMessage = "Error while retrieving throttling policy. Name : " + policyName;
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        }
+        return null;
+    }
+
+    /**
+     * Returns the ExportThrottlePolicyDTO by reading the file from input stream
+     *
+     * @param uploadedInputStream Input stream from the REST request
+     * @param fileDetail          Details of the file received via InputStream
+     * @return ExportThrottlePolicyDTO of the file to be imported
+     */
+    public static ExportThrottlePolicyDTO getImportedPolicy(InputStream uploadedInputStream, Attachment fileDetail)
+            throws ParseException, APIImportExportException, IOException {
+        File importFolder = CommonUtil.createTempDirectory(null);
+        String uploadFileName = fileDetail.getContentDisposition().getFilename();
+        String fileType = (uploadFileName.contains(ImportExportConstants.YAML_EXTENSION)) ?
+                PolicyConstants.EXPORT_POLICY_TYPE_YAML :
+                PolicyConstants.EXPORT_POLICY_TYPE_JSON;
+        String absolutePath = importFolder.getAbsolutePath() + File.separator + uploadFileName;
+        File targetFile = new File(absolutePath);
+        FileUtils.copyInputStreamToFile(uploadedInputStream, targetFile);
+        return preprocessImportedArtifact(absolutePath, fileType);
+    }
+
+    /**
+     * Preprocesses the either yaml or json file into the ExportThrottlePolicyDTO
+     *
+     * @param absolutePath temporary location of the throttle policy file
+     * @param fileType     Type of the file to be imported (.yaml/.json)
+     * @return ExportThrottlePolicyDTO from the file
+     */
+    private static ExportThrottlePolicyDTO preprocessImportedArtifact(String absolutePath, String fileType)
+            throws IOException, ParseException {
+        ExportThrottlePolicyDTO importPolicy;
+        FileReader fileReader = new FileReader(absolutePath);
+        if (Objects.equals(fileType, PolicyConstants.EXPORT_POLICY_TYPE_YAML)) {
+            ObjectMapper yamlMapper = new ObjectMapper(new YAMLFactory());
+            importPolicy = yamlMapper.readValue(fileReader, ExportThrottlePolicyDTO.class);
+        } else {
+            ObjectMapper mapper = new ObjectMapper();
+            JSONParser parser = new JSONParser();
+            Object obj;
+            obj = parser.parse(fileReader);
```

Comment:
declaration and initialization can be done at the same time

Reply:
Fixed with 30d0ecb9c29c0024fa182ba597c8eda4c128c451